extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -115,6 +116,30 @@ export class Bind {
   }
 
   /**
+   * Same as `setState()` except this method supports keys nested with '.'.
+   * E.g. `{'foo.bar': '123'}` will be transformed into `{foo: {bar: 123}}`.
+   * @param {!Object} state
+   */
+  setStateWithNestedKeys(state) {
+    const unnestedState = map();
+
+    Object.keys(state).forEach(key => {
+      let value = state[key];
+      const fragments = key.split('.');
+      // Create a wrapping object for each '.' in key, starting from inside.
+      // Stop before the first fragment, which is the top-most key.
+      for (let i = fragments.length - 1; i >= 1; i--) {
+        const wrapper = map();
```

Comment:
Doesn't this overwrite the same keys in the same input  map?

E.g. base has two keys `p.a` and `p.b`. It'd seem that this algorithm would first create object `{p: {a: 1}}` and then overwrite it with `{p: {b: 2}}`? In other words, shouldn't this line be something like:
```
const wrapper = untestedState[fragments[i]] || map();
```
?

And further, should we continue overriding the internal state the same way? Or merge?